modules/text_server_adv/thorvg_svg_in_ot.cpp

```diff
@@ -91,57 +138,100 @@ FT_Error tvg_svg_in_ot_preset_slot(FT_GlyphSlot p_slot, FT_Bool p_cache, FT_Poin
 		parser->_open_buffer((const uint8_t *)document->svg_document, document->svg_document_length);
 
 		String xml_body;
+
 		double embox_x = document->units_per_EM;
 		double embox_y = document->units_per_EM;
-		while (parser->read() == OK) {
-			if (parser->has_attribute("id")) {
-				const String &gl_name = parser->get_named_attribute_value("id");
-				if (gl_name.begins_with("glyph")) {
-					int dot_pos = gl_name.find_char('.');
-					int64_t gl_idx = gl_name.substr(5, (dot_pos > 0) ? dot_pos - 5 : -1).to_int();
-					if (p_slot->glyph_index != gl_idx) {
-						parser->skip_section();
-						continue;
-					}
-				}
-			}
-			if (parser->get_node_type() == XMLParser::NODE_ELEMENT && parser->get_node_name() == "svg") {
-				if (parser->has_attribute("viewBox")) {
-					PackedStringArray vb = parser->get_named_attribute_value("viewBox").split(" ");
 
-					if (vb.size() == 4) {
-						embox_x = vb[2].to_float();
-						embox_y = vb[3].to_float();
+		if (!state->document_map.has(document->svg_document)) {
+			state->document_map[document->svg_document] = TVG_DocumentCache();
+		}
+		TVG_DocumentCache &cache = state->document_map[document->svg_document];
+
+		if (!cache.xml_body.is_empty()) {
+			// If we have a cached document, that means we have already parsed it.
+			// All node cache should be available.
+
+			xml_body = cache.xml_body;
+			embox_x = cache.embox_x;
+			embox_y = cache.embox_y;
+
+			DEV_ASSERT(cache.node_caches.has(p_slot->glyph_index));
```

Comment:
`DEV_ASSERT` will crash, it's too aggressive for emoji parsing, something like **`ERR_FAIL_COND_V(!cache.node_caches.has(p_slot->glyph_index), FT_Err_Invalid_SVG_Document);`** should be used instead.